packages/logging/src/logging-and-telemetry.ts

```diff
@@ -0,0 +1,875 @@
+        properties,
+      });
+    });
+
+    onBus('mongosh:start-session', (args: SessionStartedEvent) => {
+      const normalizedTimingsArray = Object.entries(args.timings).map(
+        ([key, duration]) => {
+          const snakeCaseKey = toSnakeCase(key);
+          return [snakeCaseKey, duration];
+        }
+      );
+
+      const normalizedTimings = Object.fromEntries(normalizedTimingsArray);
+      this.analytics.track({
+        ...getTelemetryUserIdentity(),
+        event: 'Startup Time',
+        properties: {
+          ...getTrackingProperties(),
+          is_interactive: args.isInteractive,
+          js_context: args.jsContext,
+          ...normalizedTimings,
+        },
+      });
+    });
+
+    onBus(
+      'mongosh:new-user',
+      (newTelemetryUserIdentity: { userId: string; anonymousId: string }) => {
+        if (!newTelemetryUserIdentity.anonymousId) {
+          this.busEventState.userId = newTelemetryUserIdentity.userId;
+        }
+        this.busEventState.telemetryAnonymousId =
+          newTelemetryUserIdentity.anonymousId;
+        this.analytics.identify({
+          anonymousId: newTelemetryUserIdentity.anonymousId,
+          traits: getUserTraits(),
+        });
+      }
+    );
+
+    onBus(
+      'mongosh:update-user',
+      (updatedTelemetryUserIdentity: {
+        userId: string;
+        anonymousId?: string;
+      }) => {
+        if (updatedTelemetryUserIdentity.anonymousId) {
+          this.busEventState.telemetryAnonymousId =
+            updatedTelemetryUserIdentity.anonymousId;
+        } else {
+          this.busEventState.userId = updatedTelemetryUserIdentity.userId;
+        }
+        this.analytics.identify({
+          ...getTelemetryUserIdentity(),
+          traits: getUserTraits(),
+        });
+        this.log.info(
+          'MONGOSH',
+          mongoLogId(1_000_000_005),
+          'config',
+          'User updated'
+        );
+      }
+    );
+
+    onBus('mongosh:error', (error: Error, context: string) => {
+      const mongoshError = error as {
+        name: string;
+        message: string;
+        code: unknown;
+        scope: unknown;
+        metadata: unknown;
+      };
+
+      if (this.log) {
+        this.log[context === 'fatal' ? 'fatal' : 'error'](
+          'MONGOSH',
+          mongoLogId(1_000_000_006),
+          context,
+          `${mongoshError.name}: ${mongoshError.message}`,
+          error
+        );
+      }
+
+      if (error.name.includes('Mongosh')) {
+        this.analytics.track({
+          ...getTelemetryUserIdentity(),
+          event: 'Error',
+          properties: {
+            ...getTrackingProperties(),
+            name: mongoshError.name,
+            code: mongoshError.code,
+            scope: mongoshError.scope,
+            metadata: mongoshError.metadata,
+          },
+        });
+      }
+    });
+
+    onBus('mongosh:write-custom-log', (event: WriteCustomLogEvent) => {
+      if (this.log) {
```

Comment:
Is this still necessary? If not, we can remove it, if it is, we can still simply this with `?.`